### ISSUE TITLE
feat[dev]: ✨ implement A11y improvements to sd-button

### ISSUE THREAD

**Issue by yoezlem** [solid-design-system/solid#1469]
## Description
This task describes the Storybook implementation for the sd-button component as part of the A11y improvements outlined in Epic https://github.com/solid-design-system/solid/issues/1465.

A11y improvements for sd-button can be found here: [sd-button Notes](https://www.notion.so/matuzo/sd-button-104f3f6e1bc0800b9f72d2a046316ec3).

-----------------------------------------

## Comment
**Problem:** The accessible name of the loading buttons is “Loading Loading.” It’s coming from the slotted “Loading” and the accessible name of the progress bar.
**Solution:** The `progressbar` role needs an accessible name, so you can’t remove that. You can’t and probably shouldn’t remove the slotted content. I guess putting `aria-labelledby` (or `aria-label`) on the button to overwrite the accessible name coming for the content within is the best solution.

![Image](https://github.com/user-attachments/assets/28521cf3-d7de-4199-8e64-33a75c966464)

**Problem:** The [icon-only](https://solid-design-system.fe.union-investment.de/x.x.x/storybook/?path=/story/components-sd-button-screenshots-sd-button--icon-only) buttons have no accessible name.
**Solution:** Provide an accessible name via slotted content, coming from the icon (needs a `role=img` for that), or `aria-label`.

![Image](https://github.com/user-attachments/assets/a8a0757b-cc9b-459d-b0a9-6d521f200b1f)

## Notes
I’m not the biggest fan of disabled buttons because they cause more problems for users than benefits.
- Missing Feedback - When you click a disabled button, nothing happens. The button doesn’t explain what’s wrong or help you fix the problem.
- Missing focus - Disabled buttons are not focusable, so screen-reader users who use the Tab key to navigate might not know that there even is a button.
- Low contrast - WCAG’s minimum contrast rule doesn’t apply to disabled controls, but they’re often hard to read, especially for people with low vision
- Deception - It’s not always apparent that buttons are disabled. Some users will try to click them; if nothing happens, they can feel irritated, confused, or disappointed.

- The pattern description says, "Buttons perform various functions (e.g., download, link)." While a button can do that, aren't those examples something a link typically does?
- [ ] please check out https://shoelace.style/components/button#link-buttons
------ 
- [ ] we will create a new Ticket for the disabled topic to discuss the usage of that (design wise)

## Best Practices
The button element has two main use cases: submitting a form, and running Java‐Script when a user interacts with it. Don’t use button to link to other pages.

Always provide an label even if it’s an icon-only button.

Try to avoid the use of disabled buttons. Instead of disabling buttons, there are several measures you can take to avoid errors up front:
- Use clear labels for your input fields.
- Add hints and descriptions when the label alone isn’t clear enough.
- Split complex forms into multiple steps or pages to reduce cognitive load.
- Always enable buttons and validate input on submit.
- Give clear error messages.

## Related WCAG criteria
https://www.w3.org/WAI/WCAG22/quickref/#name-role-value
 

### Open Questions
- [ ] Question1
- [ ] Question2

## DoR
- [x] Item has been estimated by the team
- [x] Item dependencies have been identified and documented

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch
